### ISSUE TITLE
Blocks incoming PM's from highlighted friends

### ISSUE THREAD

**Issue by FrayedOP** [wtommyw/pm-highlight#5]
Can send PM's to friend as they are on my friends list, however when they reply back, it is not shown to me. Right clicking on the persons name in pm's shows add friend option, but when selecting it, it brings up the enter message box.

Had to uninstall and restart runelite to get rid of issue, not sure why it was happening but was fixed only after uninstalling plugin and restarting runelite.

**Comment by wtommyw** [comment]
Hey @FrayedOP, is your private set to `friends` by any chance? The name highlight option (currently) only works with private set to `on`. 

This is because the name is wrapped in a set of tags to give it a colour, and thus the name does not exactly match the name in your friends list. 

I'm still looking for a solution, but I've not had much time to do so recently.

**Comment by FrayedOP** [comment]
Yes it was, that makes a lot of sense now. Thank you for the quick
response! I hope there is an easy solution to be had.

Stay safe.

**Comment by Gimmee1** [comment]
Is this still in the process of seeking a solution? I just had to figure out by this plugin's issues post as to why I was not receiving PMs last night.

Bonus note: Right-clicking PMs of already added friends shows "Add friend" and "Add ignore" rather than simply "Message"

**Comment by wtommyw** [comment]
I was secretly hoping Jagex would have released (or at least, announced a date for) their official client plugin API by now.

The issues lies with the implementation (see https://github.com/wtommyw/pm-highlight/issues/5#issuecomment-1572431958), I'll try and see if I could maybe hook into a different event but I'm not very hopeful.

I will at least add a warning in the plugin panel so this issue is more clear in the meantime.

**Comment by wtommyw** [comment]
I've investigated some more and it looks like highlighting names isn't possible when you change your privacy settings on friends chat.
